### PR TITLE
feat: add metadata to spans and traces table

### PR DESCRIPTION
**Pull request by axiomofjoy**
Adds metadata to spans and traces table.

Spans

Screenshot 2024-02-20 at 5 56 00 PM

Traces

Screenshot 2024-02-20 at 5 56 16 PM

### CODE REVIEW

File: src/phoenix/server/api/types/Span.py [hunk 1]
```diff
@@ -330,3 +333,16 @@ def _hide_embedding_vectors(
...
+def _convert_metadata_to_string(metadata: Any) -> Optional[str]:
+    """
+    Converts metadata to a string representation.
+    """
+
+    if metadata is None or isinstance(metadata, str):
+        return metadata
+    try:
+        return json.dumps(metadata)
+    except json.JSONDecodeError:
```
Comment (RogerHYang) [inline src/phoenix/server/api/types/Span.py:347]: the most likely error is actually `TypeError: Object of type object is not JSON serializable`
```suggestion
    except Exception:
```

File: app/schema.graphql [hunk 2]
```diff
@@ -624,6 +624,7 @@ type Span {
 
   """Span attributes as a JSON string"""
   attributes: String!
+  metadata: String
```
Comment (mikeldking) [inline app/schema.graphql:627]: Needs a description that it's a JSON string

**Review by mikeldking — APPROVED** [review]
lgtm

File: app/schema.graphql (hunk 2 above)
Comment (axiomofjoy) [reply app/schema.graphql:627]: addressed in 95666edf

**Pull request merged**